src/electionguard_cli/steps/import_ballots/import_ballots_input_retrieval_step.py

```diff
@@ -0,0 +1,69 @@
+from io import TextIOWrapper
+from os import listdir
+from os.path import join
+from typing import List
+from click import echo
+from electionguard import CiphertextElectionContext
+
+from electionguard.ballot import SubmittedBallot
+from electionguard.guardian import Guardian, PrivateGuardianRecord
+from electionguard.manifest import Manifest
+from electionguard.serialize import from_file, from_file_wrapper, from_list_in_file
+from electionguard_cli.cli_models.import_ballots.import_ballot_inputs import (
+    ImportBallotInputs,
+)
+from electionguard_cli.steps.shared.input_retrieval_step_base import (
+    InputRetrievalStepBase,
+)
+
+
+class ImportBallotsInputRetrievalStep(InputRetrievalStepBase):
+    """Responsible for retrieving and parsing user provided inputs for the CLI's import ballots command."""
+
+    def get_inputs(
+        self,
+        manifest_file: TextIOWrapper,
+        context_file: TextIOWrapper,
+        ballots_dir: str,
+        guardian_keys: str,
+    ) -> ImportBallotInputs:
+
+        self.print_header("Retrieving Inputs")
+        manifest: Manifest = self._get_manifest(manifest_file)
+        context = self._get_context(context_file)
+        guardians = ImportBallotsInputRetrievalStep._get_guardians(
+            guardian_keys, context
+        )
+        submitted_ballots = ImportBallotsInputRetrievalStep._get_ballots(ballots_dir)
+
+        self.print_value("Ballots Dir", ballots_dir)
+
+        return ImportBallotInputs(guardians, manifest, submitted_ballots, context)
+
+    @staticmethod
+    def _get_guardians(
```

Comment:
Here's where the magic happens.  This reconstructs each guardian including private keys from the guardian private key data that was exported when you append `--output-record="./election_record.zip"` to the e2e command.  Without something like this there would be no way to use the CLI to decrypt ballots that were encrypted in another system such as the C++ core.